src/commands/run.rs

```diff
@@ -580,42 +580,50 @@ pub async fn run(args: Args) -> Result<()> {
         &output_dir,
     );
 
-    let mut evaluate = evaluator.run(token.clone(), events).boxed();
-
-    select! {
-        // Always prefer the CTRL-C signal to the evaluation returning.
-        biased;
-
-        _ = tokio::signal::ctrl_c() => {
-            error!("execution was interrupted: waiting for evaluation to abort");
-            token.cancel();
-            let _ = evaluate.await;
-            let _ = transfer_progress.await;
-            let _ = crankshaft_progress.await;
-            bail!("execution was aborted");
-        },
-        res = &mut evaluate => {
-            let _ = transfer_progress.await;
-            let _ = crankshaft_progress.await;
-
-            match res {
-                Ok(outputs) => {
-                    println!("{}", serde_json::to_string_pretty(&outputs.with_name(&entrypoint))?);
-                    Ok(())
+    let mut evaluate = evaluator.run(cancellation.clone(), events).boxed();
+
+    loop {
+        select! {
+            // Always prefer the CTRL-C signal to the evaluation returning.
+            biased;
+
+            _ = tokio::signal::ctrl_c() => {
+                // If the cancellation token was canceled, the user doesn't want to wait for evaluation to complete
+                if cancellation.token().is_cancelled() {
```

Comment:
nit: should this be `cancellation.is_cancelled()` to handle the case where an error has occurred and we want to interrupt it from waiting for the tasks to finish without having to press Ctrl+C twice? This is also a comment that the line just semantically could be clearer.